src/operations/either.jl

```diff
@@ -181,6 +181,14 @@ function applyeager(op::Either, img::AbstractArray, idx)
     _offsetarray(applyeager(op.operations[idx], img))
 end
 
+# specialize for Either to preserve type stability
+@inline function applyeager(op::Either, img::AbstractArray)
+    applyeager(op, img, randparam(op, img))
+end
+@inline function applyview(op::Either, img::AbstractArray)
+    applyview(op, img, randparam(op, img))
+end
```

Comment:
The general definition is in `operation.jl`

```julia
for FUN in (:applyeager, :applylazy, :applypermute,
            :applyaffine, :applyaffineview,
            :applyaffine_common, :applyaffineview_common,
            :applyview, :applystepview)
    @eval begin
        function ($FUN)(op::Operation, imgs::Tuple)
            param = randparam(op, imgs)
            map(img -> ($FUN)(op, img, param), imgs)
        end
        @inline function ($FUN)(op::Operation, img::AbstractArray)
            ($FUN)(op, img, randparam(op, img))
        end
    end
end
```

But the inferred type for `Either` will be `Any`(without  using `_offsetarray`) or `OffsetArray`. Add a specialization here tells the compiler more type information.